pkg/kv/kvserver/batcheval/cmd_delete_range.go

```diff
@@ -85,8 +86,28 @@ func DeleteRange(
 			args.Key, args.EndKey, desc.StartKey.AsRawKey(), desc.EndKey.AsRawKey())
 		maxIntents := storage.MaxIntentsPerWriteIntentError.Get(&cArgs.EvalCtx.ClusterSettings().SV)
 
-		err := storage.MVCCDeleteRangeUsingTombstone(ctx, readWriter, cArgs.Stats,
-			args.Key, args.EndKey, h.Timestamp, cArgs.Now, leftPeekBound, rightPeekBound, maxIntents)
+		if args.Predicates == nil {
+			err := storage.MVCCDeleteRangeUsingTombstone(ctx, readWriter, cArgs.Stats,
+				args.Key, args.EndKey, h.Timestamp, cArgs.Now, leftPeekBound, rightPeekBound, maxIntents)
+			return result.Result{}, err
+		}
+		maxBatchSize := h.MaxSpanRequestKeys
+		if h.MaxSpanRequestKeys == 0 {
+			maxBatchSize = math.MaxInt64
+		}
+
+		// The minimum number of keys required in a run to use a range tombstone
+		// TODO (msbutler): how to tune this?
+		defaultRangeTombstoneThreshold := int64(64)
```

Comment:
any advice on how to tune this?

Reply:
This is going to come down to two things: how expensive is it to write them vs. point tombstones, and how expensive is it to read them? We should benchmark this, but I'd put a `TODO` for now until we've optimized these operations, since they're currently likely to be much slower than their final form. 64 seems ok for now.

Reply:
Done. Added Todo.